File: connectors/connector-redmine/src/main/scala/com/taskadapter/connector/redmine/FieldRowFinder.scala
package com.taskadapter.connector.redmine

import com.taskadapter.connector.FieldRow
import com.taskadapter.model.Field

object FieldRowFinder {
  // TODO maybe fieldsRows.exists(_.exists(f => f.name.equals(field.name))) ?
  def containsTargetField(fieldRows: Seq[FieldRow[_]], field: Field[_]): Boolean = {
    fieldRows.foreach { row =>
      val target = row.targetField
      if (target.isDefined && target.get.name.equals(field.name)) {
        return true
      }
    }
    false
  }
}

File: connectors/connector-redmine/src/main/java/com/taskadapter/connector/redmine/FieldRowFinder.java
package com.taskadapter.connector.redmine;

import com.google.common.collect.Lists;
import com.taskadapter.connector.FieldRow;
import com.taskadapter.model.Field;

import java.util.Optional;

public class FieldRowFinder {
    // TODO 14 move to common?
    public static boolean containsTargetField(Iterable<FieldRow<?>> fieldRows, Field<?> field) {
        return Lists.newArrayList(fieldRows).stream().anyMatch(row -> {
            Optional<? extends Field<?>> target = row.targetField();
            return target.isPresent()
                    && target.get().getClass().getName().equals(field.getClass().getName());
        });
    }
}
